### PR TITLE
[PDI-16369]-Opening a transformation or job uploaded to the repository using REST API results in "no transformer found" exception

### PR DESCRIPTION
**Pull request by denisprotopopov**
- added fix for transformations and jobs

### CODE REVIEW

**Comment by denisprotopopov** [comment]
@mbatchelor please review the changes

**Comment by denisprotopopov** [comment]
@pamval @mbatchelor please review the changes

File: repository/src/main/java/org/pentaho/platform/repository2/unified/fileio/RepositoryFileOutputStream.java [hunk 1]
```diff
@@ -179,7 +181,18 @@ public void flush() throws IOException {
     // RepositoryFileWriter
     // but I couldn't figure out a clean way to do that. For now, charsetName is passed in here and we use it if
     // available.
-    final SimpleRepositoryFileData payload = new SimpleRepositoryFileData( bis, charsetName, mimeType );
+    final IRepositoryFileData payload;
+    Converter converter;
+
+    if ( "ktr".equalsIgnoreCase( ext ) ) {
+      converter = PentahoSystem.get( Converter.class, null, Collections.singletonMap( "name", "PDITransformationStreamConverter" ) );
```
Comment (pamval) [inline repository/src/main/java/org/pentaho/platform/repository2/unified/fileio/RepositoryFileOutputStream.java]: I think this code will break in CE. Can you be a bit more defensive and test if converter is null for both ktr and kjb?

If it is null, then just default to new SimpleRepositoryFileData as before

Comment (pamval) [inline repository/src/main/java/org/pentaho/platform/repository2/unified/fileio/RepositoryFileOutputStream.java]: Also, a unit or integration test would be nice. Not sure if it is possible, but would be nice.

Comment (denisprotopopov) [inline repository/src/main/java/org/pentaho/platform/repository2/unified/fileio/RepositoryFileOutputStream.java]: @pamval please review the changes
